aiomcache/pool.py

```diff
@@ -72,26 +61,24 @@ def release(self, conn):
         :param conn: ``namedtuple`` (reader, writer)
         """
         self._in_use.remove(conn)
-
         if conn.reader.at_eof() or conn.reader.exception():
             self._do_close(conn)
         else:
-            # This should never fail because poolsize=maxsize
             self._pool.put_nowait(conn)
 
     @asyncio.coroutine
     def _create_new_conn(self):
-        if self._size < self._maxsize:
-            self._size += 1
-            try:
-                reader, writer = yield from asyncio.open_connection(
-                    self._host, self._port, loop=self._loop)
-            except:
-                self._size -= 1
-                raise
-            return _connection(reader, writer)
+        if self.size() < self._maxsize:
+            reader, writer = yield from asyncio.open_connection(
+                self._host, self._port, loop=self._loop)
+            if self.size() < self._maxsize:
```

Comment:
Avoid race condition when yielding from `open_connection`. When we get a new connection, maybe other clients also created their connections so we have to check again and close it if necessary